### PR TITLE
[INJIMOB-890]: add workflow to publish android artifact to maven

### PR DESCRIPTION
**Pull request by AyushiDey**

### CODE REVIEW

File: .github/workflows/maven-publish-android.yml [hunk 1]
```diff
@@ -0,0 +1,142 @@
+name: Maven build
+
+on:
+  workflow_dispatch:
```
Comment (syedsalman3753) [inline .github/workflows/maven-publish-android.yml:4]: replace `workflow_dispatch` with `workflow_call`

File: .github/workflows/maven-publish-android.yml [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+        description: 'GPG secret'
+        required: true
+      SLACK_WEBHOOK_URL:
+        description: 'Slack webhook URL'
+        required: true
+    branches:
+      - release*
+      - master
+      - develop*
+
```
Comment (syedsalman3753) [inline .github/workflows/maven-publish-android.yml:39]: Please remove this as it is not required

File: .github/workflows/maven-publish-android.yml [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+        required: true
+      OSSRH_TOKEN:
+        description: 'OSSRH token'
+        required: true
+      GPG_SECRET:
+        description: 'GPG secret'
+        required: true
+      SLACK_WEBHOOK_URL:
+        description: 'Slack webhook URL'
+        required: true
```
Comment (syedsalman3753) [inline .github/workflows/maven-publish-android.yml:34]: these has to be github secrets

```
    secrets:
      OSSRH_USER:
        required: true
      OSSRH_SECRET:
        required: true
      OSSRH_TOKEN:
        required: true
      GPG_SECRET:
        required: true
      SLACK_WEBHOOK_URL:
        required: true
```

File: .github/workflows/maven-publish-android.yml [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+          distribution: adopt
+          java-version: 11
+          server-id: ossrh # Value of the distributionManagement/repository/id field of the pom.xml
+          settings-path: ${{ github.workspace }} # location for the settings.xml file
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: '16'
+
```
Comment (syedsalman3753) [inline .github/workflows/maven-publish-android.yml:57]: please add below cache for gradle and npm 

```
      - name: Cache Gradle packages
        uses: actions/cache@v3.3.1
        with:
          path: |
            ~/.gradle/caches
            ~/.gradle/wrapper
          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle') }}
          restore-keys: ${{ runner.os }}-gradle-

      - name: Cache npm dependencies
        uses: actions/cache@v2
        with:
          path: '~/.npm'
          key: ${{ runner.os }}-npm-${{ hashFiles('**/package-lock.json') }}
          restore-keys: |
            ${{ runner.os }}-npm-
```

File: .github/workflows/maven-publish-android.yml [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+      ANDROID_SERVICE_LOCATION:
+        description: "Android Service Location"
+        required: true
+        default: "android"
+        type: string
+      BUILD_ARTIFACT:
+        description: 'Build artifact'
+        required: true
+        type: string
+        default: 'artifact'
```
Comment (syedsalman3753) [inline .github/workflows/maven-publish-android.yml:19]: please remove this as it is not used in the workflow

**Review by syedsalman3753 — CHANGES_REQUESTED** [review]

**Review by syedsalman3753 — APPROVED** [review]

**Review by ckm007 — APPROVED** [review]

**Pull request merged**